src/main/java/com/faforever/api/clan/ClanService.java

```diff
@@ -93,12 +99,14 @@ String generatePlayerInvitationToken(Player requester, int newMemberId, int clan
       throw new ApiException(new Error(ErrorCode.CLAN_NOT_LEADER, clanId));
     }
 
-    Player newMember = playerRepository.findById(newMemberId)
-      .orElseThrow(() -> new ApiException(new Error(ErrorCode.CLAN_GENERATE_LINK_PLAYER_NOT_FOUND, newMemberId)));
+    Player newMember = clan.getIsOpen() ? null :
```

Comment:
What should happen if the clan is not open.?For me it seems like there should not be an invite link generated, if so please throw a custom ApiException

Reply:
I mean if newMember if null and expire is 0 that seems weird to me.